java-checks/src/main/java/org/sonar/java/checks/tests/MockitoAnnotatedObjectsShouldBeInitializedCheck.java

```diff
@@ -73,16 +79,32 @@
   @Override
   public void visitNode(Tree tree) {
     ClassTree testClass = (ClassTree) tree;
+    if (visited.contains(testClass)) {
+      return;
+    }
+
+    if (isClassProperlyAnnotated(testClass)) {
+      List<ClassTree> children = getChildrenCoveredByAnnotations(testClass);
+      visited.addAll(children);
```

Comment:
You can probably return here, and remove the redundant `isClassProperlyAnnotated` computation in `mocksAreProperlyInitialized`.

Reply:
Modified